platooning_strategic_IHP/include/platoon_strategic_ihp.h

```diff
@@ -149,6 +150,21 @@ namespace platoon_strategic_ihp
             */
             cav_msgs::Maneuver composeManeuverMessage(double current_dist, double end_dist, double current_speed, double target_speed, int lane_id, ros::Time& current_time);
 
+            /**
+            * \brief Find start(current) and target(end) lanelet index from path to generate lane change maneuver message.
+            * 
+            * \param current_dist current downtrack distance
+            * \param end_dist ending downtrack distance
+            * \param current_speed current speed
+            * \param target_speed target speed
+            * \param starting_lane_id current lanelet id which serves as the starting lanlet id
```

Comment:
Using current distance and current lanelet ID implies that we are constraining ourselves to planning something that must happen immediately.  Is there any chance that we may want to plan a maneuver that starts sometime in the future, in which case the host vehicle will be on a different lanelet than the current one?  Strategic plugins will be called on to make plans several seconds in the future, so it needs to handle such a situation.  I haven't studied the code enough to know if this particular method is affected, but I guess it would be.  Even if it is intended to be for "immediate" use only, consider the case where this method is called just before the vehicle rolls off the end of the current lanelet, and by the time processing time is complete the wheels are on a new lanelet.  Therefore, I think that use of "current" in these param names is wrong.